### PR TITLE
Introduction of HATEOAS on the API

### LINKED ISSUE #134: API 2.0.0
The pod API endpoint should include an "created by" information, which could have values such as "deployment", "daemonset" and others ...

### PR DESCRIPTION
**Pull request by troeger**
In an attempt to fix #134, I propose to move the API towards the [HATEOAS](https://en.wikipedia.org/wiki/HATEOAS) principle. In short this means that the frontend no longer converts UIDs returned by the API into resource URLs. Instead, the API returns such URLs by itself.

This leads to the following changes:

- Another modification of the /login/ result (sorry). It now contains the JWT tokens and the link to the user resource.
- New API endpoints for representing namespaces and service accounts as API resources.
- The user resource now contains links to portal groups, web applications, namespaces, and service accounts as links. There is still an additional flat list of namespace names for convenience.

If you agree on the general concept, than I would solve #134 based on additional resource links, meaning that deployments link to pods, and the other way around ....

image

### CODE REVIEW

File: kubeportal/tests/test_api.py [hunk 1]
```diff
@@ -60,17 +59,16 @@ def test_api_login(api_client_anon, admin_user):
...
-    assert 5 == len(data)
-    assert 'group_ids' in data
-    assert 'webapp_ids' in data
-    assert 'user_id' in data
+    assert 3 == len(data)
+    assert 'user' in data
     assert 'access_token' in data
-    assert 'k8s_namespace' in data
-    assert data['user_id'] == admin_user.pk
+    assert 'refresh_token' in data
```
Comment (DenizCankurtaran) [inline kubeportal/tests/test_api.py:65]: Why does the login route deliver both, access and refresh token?

**Review by DenizCankurtaran — COMMENTED** [review]
I think it's a really  interesting approach, which I haven't seen before.

File: kubeportal/tests/test_api.py (hunk 1 above)
Comment (troeger) [reply kubeportal/tests/test_api.py:65]: The access token is short-lived, an when it expires, you can use the refresh token to get a new one:

https://django-rest-framework-simplejwt.readthedocs.io/en/latest/getting_started.html#usage
https://auth0.com/blog/refresh-tokens-what-are-they-and-when-to-use-them/

**Comment by troeger** [comment]
The following additional routes now implement the new concept:

- /namespace/<name>/deployments/ => List of deployment links
- /deployment/<uid>/ => List of pods

Please give it a try before I continue. This is a lot of work.

**Pull request merged**